test-project/Assets/EditmodeTests/Utility/InterestBuilderTests.cs

```diff
@@ -0,0 +1,154 @@
+using System.Collections.Generic;
+using Improbable.Gdk.QueryBasedInterest;
+using NUnit.Framework;
+
+namespace Improbable.Gdk.EditmodeTests.Utility
+{
+    [TestFixture]
+    public class InterestBuilderTests
+    {
+        private static ComponentInterest.Query BasicQuery
+            => InterestQuery.Query(Constraint.RelativeSphere(10));
+
+        private static ComponentInterest.Query DifferentBasicQuery
+            => InterestQuery.Query(Constraint.RelativeSphere(20));
+
+        private static InterestBuilder BasicInterest
+            => InterestBuilder.Begin();
+
+        private static InterestBuilder ModifyInterest
+            => InterestBuilder.Modify(new Dictionary<uint, ComponentInterest>());
+
+        [Test]
+        public void AddQueries_can_be_called_multiple_times_on_same_component()
+        {
+            Assert.DoesNotThrow(() => BasicInterest
+                .AddQueries<Position.Component>(BasicQuery, BasicQuery)
+                .AddQueries<Position.Component>(BasicQuery, BasicQuery));
+        }
+
+        [Test]
+        public void AddQueries_can_be_called_after_AddQueries()
+        {
+            Assert.DoesNotThrow(() => BasicInterest
+                .AddQueries<Position.Component>(BasicQuery, BasicQuery)
+                .AddQueries<Metadata.Component>(BasicQuery, BasicQuery));
+        }
+
+        [Test]
+        public void AddQueries_can_be_called_with_a_single_query()
+        {
+            Assert.DoesNotThrow(() => BasicInterest
+                .AddQueries<Position.Component>(BasicQuery));
+        }
+
+        [Test]
+        public void ReplaceQueries_clears_previous_query()
+        {
+            var initialQuery = BasicQuery;
+            double initialQueryRadius = 0;
+            Assert.DoesNotThrow(() =>
+            {
+                initialQueryRadius = initialQuery.Constraint.RelativeSphereConstraint.Value.Radius;
+            });
+
+            var differentBasicQuery = DifferentBasicQuery;
+            double differentQueryRadius = 0;
+            Assert.DoesNotThrow(() =>
+            {
+                initialQueryRadius = DifferentBasicQuery.Constraint.RelativeSphereConstraint.Value.Radius;
+            });
```

Comment:
Why are we testing if accessing these properties throws?

Reply:
my initial thought was that this property should exist in the Constraint - but really it's not the InterestBuilder tests' job to test this